website/server/controllers/api-v3/groups.js

```diff
@@ -402,6 +408,14 @@ api.leaveGroup = {
     }
 
     await group.leave(user, req.query.keep);
+
+    _removeMessagesFromMember(user, group._id);
+
+    await Bluebird.all([
+      user.save(),
+      group.save(),
```

Comment:
What has changed in the group that requires the group to be saved? Doesn't `group.leave` handle the save logic?